### PR TITLE
Add Epic FHIR pre-built services

### PR DESCRIPTION
**Pull request by sameeragunarathne**

### CODE REVIEW

File: ehr-connectivity/epic-fhirr4-administration-api-service/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+### Setup and run
+
+1. Clone this repository to your local machine and navigate to the pre-built service on Epic administration API.
+
+2. Set the following values from environment variables.
+    - `EPIC_FHIR_SERVER_URL` - The URL of the Epic FHIR server.
+    - `EPIC_FHIR_SERVER_TOKEN_URL` - The URL of the Epic FHIR server token endpoint.
+    - `EPIC_FHIR_APP_CLIENT_ID` - The client ID of the Epic FHIR application.
+    - `EPIC_FHIR_APP_PRIVATE_KEY` - The private key created for the Epic FHIR application.
```
Comment (nirmal070125) [inline ehr-connectivity/epic-fhirr4-administration-api-service/README.md:17]: Path to the private key right?

File: ehr-connectivity/epic-fhirr4-administration-api-service/Ballerina.toml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+[[dependency]]
+org = "ballerinax"
+name = "health.fhirr4"
+version = "1.0.8"
+
+[[dependency]]
+org = "ballerinax"
+name = "health.clients.fhir"
+version = "1.0.4"
+repository="local"
```
Comment (isuruh15) [inline ehr-connectivity/epic-fhirr4-administration-api-service/Ballerina.toml:21]: Do we need to point to the local repository here?

File: ehr-connectivity/epic-fhirr4-administration-api-service/service.bal [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+        r4:Bundle|error fhirInteractionResult = executeFhirInteraction("Organization", fhirContext, (), (), uscore:USCoreOrganizationProfile).ensureType(r4:Bundle);
+        if fhirInteractionResult is error {
+            return r4:createFHIRError("Error occurred while executing the Organization search interaction.", r4:CODE_SEVERITY_ERROR,
+                r4:TRANSIENT_EXCEPTION, cause = fhirInteractionResult);
+        }
+        return fhirInteractionResult;
+    }
+
+    // Create a new resource with a server assigned id.
+    isolated resource function post fhir/r4/Organization(r4:FHIRContext fhirContext, uscore:USCoreOrganizationProfile payload) returns @http:Payload {mediaType: ["application/fhir+json"]} Organization|r4:FHIRError {
```
Comment (isuruh15) [inline ehr-connectivity/epic-fhirr4-administration-api-service/service.bal:113]: Are we expecting only the USCore resources here or developer can send resources from custom packages?

isolated resource function post fhir/r4/Organization(r4:FHIRContext fhirContext, **uscore:USCoreOrganizationProfile** payload)

Comment (sameeragunarathne) [reply ehr-connectivity/epic-fhirr4-administration-api-service/service.bal:113]: It should be us core resources for the Epic EHR. This is added like this due to a issue on the fhir service type accepting the union types

File: ehr-connectivity/epic-fhirr4-administration-api-service/Ballerina.toml (hunk 2 above)
Comment (sameeragunarathne) [reply ehr-connectivity/epic-fhirr4-administration-api-service/Ballerina.toml]: Referred the latest release fhir client on the dev.

File: ehr-connectivity/epic-fhirr4-administration-api-service/README.md (hunk 1 above)
Comment (sameeragunarathne) [reply ehr-connectivity/epic-fhirr4-administration-api-service/README.md]: Fixed the naming

**Review by isuruh15 — APPROVED** [review]

**Review by isuruh15 — APPROVED** [review]

**Pull request merged**